### PR TITLE
net: http: create empty testsuites for gsoc 2023 project

### LINKED ISSUE #59028: gsoc2023: net: http: set up test for continuous integration
```shell
mkdir ~/zephyrproject/
cd ~/zephyrproject
git clone https://github.com/zephyrproject-rtos/zephyr.git zephyr
cd zephyr
# use the GitHub web interface to fork the zephyr project
git remote add Emna-Rekik git@github.com:Emna-Rekik/zephyr.git
git fetch -p Emna-Rekik
# create a new branch from `main`
git checkout -b set-up-ci

mkdir -p tests/net/http
for i in prototype rest crime json tls; do
  mkdir -p tests/net/http/$i/src/
  touch tests/net/http/$i/CMakeLists.txt
  touch tests/net/http/$i/src/main.c
  touch tests/net/http/$i/prj.conf
  touch tests/net/http/$i/testcase.yaml
done

git add tests/net/http
git commit -s -m 'net: http: create empty testsuites for gsoc 2023 project'
git push Emna-Rekik
```

### PR DESCRIPTION
**Pull request by Emna-Rekik**
This commit adds an empty test suite for http server.
No test cases have been added in this commit.
However, creating the empty test suite provides a structure for adding tests in subsequent commits.

Fixes #59028

### CODE REVIEW

File: tests/net/http/crime/testcase.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+tests:
+  # section.subsection
+  testing.ztest:
+    build_only: true
+    platform_allow: native_posix
+    integration_platforms:
+      - native_posix
```
[the remark this answers is not in the document]
Comment (cfriedt) [reply tests/net/http/crime/testcase.yaml:7]: And maybe repeat for the other test suites.

**Comment by cfriedt** [comment]
Aha - @Emna-Rekik - there are a few tools you can use to check your coding style prior to check in as well. I would say, #1 is `checkpatch.pl`. I use it in "git mode" mostly, and you just run the following after check-in N varies based on the number of commits you've made (N > 0).
```shell
./scripts/checkpatch.pl -g HEAD~N..
```

And as well, each file needs an `SPDX-license-Identifier: Apache-2.0` license in comments. Comment formats vary, but one possible license is below:
```cpp
/*
 * Copyright (c) 2023, Emna Rekik
 *
 * SPDX-License-Identifier: Apache-2.0
 */
```

Looks good though, I think it's almost ready 😀

**Comment by rlubos** [comment]
I'd put the tests in `tests/net/lib` - we even already have a `http_server` subdirectory in there, with a test suite introduced by Chris.

**Comment by rlubos** [comment]
Also, I'd squash the two commits - adding a commit with empty files doesn't bring much value and would be hard to justify (which somehow explains lack of the commit message :) ).

**Pull request closed without merging**